api/models/post/validatePostData.js

```diff
@@ -4,16 +4,23 @@ export default function validatePostData (userId, data) {
   if (!trim(data.name)) {
     throw new Error('title can\'t be blank')
   }
-  const allowedTypes = [Post.Type.REQUEST, Post.Type.OFFER, Post.Type.DISCUSSION, Post.Type.PROJECT, Post.Type.EVENT]
+
+  const allowedTypes = [Post.Type.REQUEST, Post.Type.OFFER, Post.Type.DISCUSSION, Post.Type.PROJECT, Post.Type.EVENT, Post.Type.RESOURCE]
   if (data.type && !includes(allowedTypes, data.type)) {
     throw new Error('not a valid type')
   }
+
   if (isEmpty(data.community_ids)) {
     throw new Error('no communities specified')
   }
+
   if (data.topicNames && data.topicNames.length > 3) {
     throw new Error('too many topics in post, maximum 3')
   }
+  
+  if (data.type === Post.Type.RESOURCE && !data.location_id) {
+    throw new Error('resources must have a location')
```

Comment:
I've actually been wondering recently if we might not want to use resources for other types of resources like books, links, files... that the community wants to kind of "store" or track... but thats for the future. location based ones only for now is good!